File: src/main/java/com/example/simplecounter/MyService.java
package com.example.simplecounter;

import android.app.Service;
import android.content.ContentValues;
import android.content.Intent;
import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.os.IBinder;
import android.support.annotation.Nullable;
import android.util.Log;

import java.util.ArrayList;
import java.util.Timer;
import java.util.TimerTask;

public class MyService extends Service {
    static String LOG_TAG = "serviceLogs";
    static final String ACTION = "com.example.simplecounter.display";
    Intent intent;
    Timer timer;

    @Override
    public void onCreate() {
        super.onCreate();
        Log.d(LOG_TAG, "onCreate done");
        intent = new Intent(ACTION);
    }
    @Override
    public int onStartCommand(Intent intent, int flags, int startId) {
        Log.d(LOG_TAG, "onStartCommand done");
        timer = new Timer();
        timer.schedule(new CounterTask(), 1000, 1000);
        return super.onStartCommand(intent, flags, startId);
    }
    @Override
    public void onDestroy() {
        super.onDestroy();
        timer.cancel();
        Log.d(LOG_TAG, "onDestroy done");
    }
    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        Log.d(LOG_TAG, "onBind done");
        return null;
    }

    class CounterTask extends TimerTask {
        DBHelper helper;
        SQLiteDatabase db;
        int units;

        CounterTask() {
            helper = new DBHelper(getApplicationContext());
        }

        @Override
        public void run() {
            new Thread(new Runnable() {
                @Override
                public void run() {
                    db = helper.getWritableDatabase();
                    ContentValues cv = new ContentValues();
                    cv.put(DBHelper.KEY_VALUE, units);
                    db.insert(DBHelper.TABLE_NAME, null, cv);
                    db.close();

                    Log.d(LOG_TAG, String.valueOf(units));
                    intent.putExtra("value", String.valueOf(units));
                    sendBroadcast(intent);
                }
            }).start();
            units++;
        }
    }
}

File: src/main/java/com/example/simplecounter/MyService.kt
package com.example.simplecounter

import android.app.Service
import android.content.ContentValues
import android.content.Intent
import android.database.Cursor
import android.database.sqlite.SQLiteDatabase
import android.os.IBinder
import android.util.Log

import java.util.ArrayList
import java.util.Timer
import java.util.TimerTask

class MyService : Service() {
    internal var intentValue: Intent? = null
    internal var timer: Timer? = null

    override fun onCreate() {
        super.onCreate()
        Log.d(LOG_TAG, "onCreate done")
        intentValue = Intent(ACTION_VAL)
    }

    override fun onStartCommand(intent: Intent, flags: Int, startId: Int): Int {
        Log.d(LOG_TAG, "onStartCommand done")
        timer = Timer()
        timer?.schedule(CounterTask(), 1000, 1000)
        return super.onStartCommand(intent, flags, startId)
    }

    override fun onDestroy() {
        super.onDestroy()
        timer?.cancel()
        Log.d(LOG_TAG, "onDestroy done")
    }

    override fun onBind(intent: Intent): IBinder? {
        Log.d(LOG_TAG, "onBind done")
        return null
    }

    internal inner class CounterTask : TimerTask() {
        var helper: DBHelper
        var db: SQLiteDatabase? = null
        var units: Int = 0

        init {
            helper = DBHelper(applicationContext)
        }

        override fun run() {
            Thread(Runnable {
                db = helper.writableDatabase
                val cv = ContentValues()
                cv.put(DBHelper.KEY_VALUE, units)
                db?.insert(DBHelper.TABLE_NAME, null, cv)
                db?.close()

                Log.d(LOG_TAG, units.toString())
                intentValue!!.putExtra("value", units.toString())
                sendBroadcast(intentValue)
            }).start()
            units++
        }
    }

    companion object {
        internal var LOG_TAG = "serviceLogs"
        internal val ACTION_VAL = "com.example.simplecounter.display"
    }
}
